### PR TITLE
[DLSP21] Transformer encoder-predictor-decoder

### PR DESCRIPTION
**Pull request by jingshuaijiang**

### CODE REVIEW

File: docs/en/week10/10-3.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+## The Transformer
+
+Expanding on our knowledge of attention in particular, we now interpret the fundamental building blocks of the transformer. In particular, we will take a forward pass through a basic transformer, and see how attention is used in the standard encoder-predictor-decoder paradigm.
+
+
+### Encoder-predictor-decoder architecture
+
+
+We should be familiar with the terminology of these modules from the previous lectures. 
+In the EBM latent variable architecture, we have $\vect{x}$ the observed input value which goes into a predictor. We have $\vect{y}$ which is the target value. The decoder modules is fed with a latent variable and output of the predictorn which outputs  $\tilda{y}$. $E$ is the energy function which minimises the energy between $\tilda{y}$ and $y$.
```
Comment (Atcold) [inline docs/en/week10/10-3.md:19]: `\tilde{\vect{}}`, not `tilda`. Please, run the website locally and make sure it is working fine.
Also, last `y` should be `\vect{y}`.

File: docs/en/week10/10-3.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+### Encoder-predictor-decoder architecture
+
+
+We should be familiar with the terminology of these modules from the previous lectures. 
+In the EBM latent variable architecture, we have $\vect{x}$ the observed input value which goes into a predictor. We have $\vect{y}$ which is the target value. The decoder modules is fed with a latent variable and output of the predictorn which outputs  $\tilda{y}$. $E$ is the energy function which minimises the energy between $\tilda{y}$ and $y$. 
+
+
+<center>
+<img src="{{site.baseurl}}/images/week10/10-3/ebm.png" style="zoom: 100%; background-color:#DCDCDC;" /><br>
+<b>Figure 1:</b> EBM model.
```
Comment (Atcold) [inline docs/en/week10/10-3.md:24]: Conditional EBM model. There's an `x`!

File: docs/en/week10/10-3.md [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+</center>
+
+
+
+In Autoencoder structue , we observed there is no conditional input but only a target variable. It is fed through an encoder module which transforms into a hidden representation space, forcing only the most important information through.
+
+<center>
+<img src="{{site.baseurl}}/images/week10/10-3/autoencoder.png" style="zoom: 100%; background-color:#DCDCDC;" /><br>
```
Comment (Atcold) [inline docs/en/week10/10-3.md:32]: This is wrong. Where did you get this slide from, last year?
What is `x`? What is `y`? Please, be careful 😇

File: docs/en/week10/10-3.md [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+
+
+
+<center>
+<img src="{{site.baseurl}}/images/week10/10-3/transformer.png" style="zoom: 100%; background-color:#DCDCDC;" /><br>
+<b>Figure 3:</b> Transformer Architecture.
+</center>
+
+
+In a transformer, $\vect{y}$(target sentence) is a discrete time signal, which is  fed into a unit delay module succeeded by an encoder. The unit delay here transforms $y[j] —> y[j-1]$.
```
Comment (Atcold) [inline docs/en/week10/10-3.md:48]: Missing a space after `$\vect{y}$`.
`->` is **not** LaTeX. Please, use LaTeX symbols.

File: docs/en/week10/10-3.md [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+
+
+
+## Summary
+
+A set, $\vect{x}\_1$ to $\vect{x}\_{t}$ is fed through the encoder. Using self-attention and some more blocks, an output representation, $\lbrace\vect{h}^\text{Enc}\rbrace_{i=1}^t$ is obtained, which is fed to the decoder. After applying self-attention to it, cross attention is applied. In this block, the query corresponds to a representation of a symbol in the target language $\vect{y}\_i$, and the key and values are from the source language sentence ($\vect{x}\_1$ to $\vect{x}\_{t}$). Intuitively, cross attention finds which values in the input sequence are most relevant to constructing $\vect{y}\_t$, and therefore deserve the highest attention coefficients. The output of this cross attention is then fed through another 1D-convolution sub-block, and we have $\vect{h}^\text{Dec}$. For the specified target language, it is straightforward from here to see how training will commence, by comparing $\lbrace\vect{h}^\text{Dec}\rbrace_{i=1}^t$ to some target data.
```
Comment (Atcold) [inline docs/en/week10/10-3.md:125]: Why did you copy stuff from your peers? This is not cool…
Just drop a link pointing to last year's content.

**Review by Atcold — CHANGES_REQUESTED** [review]
ONLY add new content. DO NOT copy content from last year's website. This is not cool…
Also, the notation changed from last year, so it's even wrong.

File: docs/en/week10/10-3.md (hunk 5 above)
Comment (rahuja123) [reply docs/en/week10/10-3.md:125]: Hi, Prof. sorry to do that. We deleted the before part, but this was in middle, so kept it. We have made the new changes.

**Pull request closed without merging**